vantage6/node/proxy_server.py

```diff
@@ -1,262 +1,339 @@
+
+    # Attemt to send the task to the central server
     try:
-        results = requests.get(
-            f"{url}/task/{id}/result",
-            headers={'Authorization': auth}
-        ).json()
+        headers = {'Authorization': request.headers['Authorization']}
+        response = make_request('post', 'task', data, headers=headers)
+    except Exception:
+        log.exception('post task failed')
+        return {'msg': 'Request failed, see node logs'},\
+            HTTPStatus.INTERNAL_SERVER_ERROR
 
-        unencrypted = []
-        for result in results:
-            result["result"] = server_io.cryptor.decrypt_str_to_bytes(
-                result["result"]
-            )
+    return response.json(), HTTPStatus.OK
 
-            result["result"] = bytes_to_base64s(result["result"])
-            unencrypted.append(
-                result
-            )
 
-        # log.error(unencrypted)
+@app.route('/task/<int:id>/result', methods=["GET"])
+def proxy_task_result(id: int) -> Response:
+    """
+    Obtain and decrypt all results to belong to a certain task
+
+    Parameters
+    ----------
+    id : int
+        Task id from which the results need to be obtained
+
+    Returns
+    -------
+    Response
+        Reponse from the vantage6-server
+    """
+    # We need the server io for the decryption of the results
+    server_io = app.config.get("SERVER_IO")
+    if not server_io:
+        return jsonify({'msg': 'Proxy server not initialized properly'}), 500
 
+    # Forward the request
+    try:
+        response = make_proxied_request(f"task/{id}/result")
     except Exception as e:
-        log.error("Proxyserver was unable to retrieve results (1)!")
         log.debug(e)
+        return {'msg': 'Request failed, see node logs'},\
+            HTTPStatus.INTERNAL_SERVER_ERROR
 
-    return jsonify(unencrypted)
+    # Attempt to decrypt the results. The enpoint should have returned
+    # a list of results
+    try:
+        unencrypted = []
+        for result in response.json():
+            if result['result']:
+                result["result"] = bytes_to_base64s(
+                    server_io.cryptor.decrypt_str_to_bytes(
+                        result["result"]
+                    )
+                )
 
+            # if the result is a None, there is no need to decrypt that..
+            unencrypted.append(result)
 
-@app.route('/result/<int:id>', methods=["GET"])
-def proxy_results(id):
-    """ Obtain result `id` from the server.
+    except Exception:
+        log.exception("Unable to decrypt and/or decode results, sending them "
+                      "to the algorithm...")
 
-        :param id: the id of the result to retrieve
+    return jsonify(unencrypted), HTTPStatus.OK
 
-        TODO decrypt the results
-    """
-    url = server_info()
 
-    server_io = app.config["SERVER_IO"]
+@app.route('/result/<int:id>', methods=["GET"])
+def proxy_results(id: int) -> Response:
```

Comment:
This function is very similar to `proxy_task_result`. I think it would be nice to refactor the two